triedb/pathdb/nodebuffer.go

```diff
@@ -204,6 +204,19 @@ func (b *nodebuffer) setSize(size int, db ethdb.KeyValueStore, clean *fastcache.
 	return b.flush(db, clean, id, false)
 }
 
+// allocBatch returns a database batch with pre-allocated buffer.
+func (b *nodebuffer) allocBatch(db ethdb.KeyValueStore) ethdb.Batch {
+	var metasize int
```

Comment:
Why don't you calculate it during the `commit`?

https://github.com/ethereum/go-ethereum/blob/8efb094e3541e44e05b04b0bc9f44590080d2b78/triedb/pathdb/nodebuffer.go#L105-L112

Reply:
IMO it's cleaner to do it as late as possible instead of risking the counters going out of sync with the code. Also less code/fields to keep in mind.